dataprep_ml/cleaners.py

```diff
@@ -370,16 +370,110 @@ def _get_columns_to_clean(data: pd.DataFrame, dtype_dict: Dict[str, dtype], mode
     return cleanable_columns
 
 
-def clean_timeseries(df: pd.DataFrame, tss: Dict) -> pd.DataFrame:
+def _fix_duplicates(df: pd.DataFrame, tss: dict) -> pd.DataFrame:
```

Comment:
To-Do (on me): benchmark a few large datasets and report back

Reply:
Around 10 seconds for a 10-duplicate DF with 10 million records. We can optimize if need be in the future, but for now this will do.